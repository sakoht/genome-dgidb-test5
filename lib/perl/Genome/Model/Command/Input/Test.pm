
###########################################################################

package Genome::Model::Command::Input::Test;

use strict;
use warnings;

use base 'Test::Class';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input';
}



###########################################################################

package Genome::Model::Command::Input::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;
require Genome::Model::Test;
require File::Temp;
require File::Path;

sub valid_param_sets {
    return grep { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub invalid_param_sets {
    return map { $_->{model_identifier} = $_[0]->_model->id } $_[0]->_valid_param_sets;
}

sub _model {
    my $self = shift;

    unless ( $self->{_model} ) {
        $self->{_model} = Genome::Model::Test->create_mock_model(
            type_name => 'tester',
            instrument_data_count => 0,
        )
            or confess "Can't create mock tester model.";
    }

    return $self->{_model};
}

sub _build {
    my $self = shift;

    return ($self->_model->builds)[0];
}



sub startup : Tests(startup => no_plan) {
    my $self = shift;
    
    no warnings;
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    use warnings;
    is_deeply(
        [ Genome::Model::Command->get_model_type_names ],
        [qw/ tester /],
        'Set tester type name.',
    );

    $self->_startup or confess "Failed startup.";
    
    return 1;
}

sub _startup {
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Add::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Add';
}

sub _valid_param_sets {
    return (
        { # add multiple 'value_ids' 
            name => 'friends',
            ids => 'Watson,Crick',
            before_execute => '_before_execute',
            after_execute => '_after_execute',
        },
        { # add single 'value'
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
            before_execute => '_before_execute',
            after_execute => '_after_execute',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to add a not 'is_many' input
            name => 'coolness', 
            ids => 'none',
        },
        { # try to add Crick again
            name => 'friends',
            ids => 'Crick',
        },
        { # try to add id that don't exist
            name => 'inst_data',
            ids => '2sep09.934noexist',
        },
    );
}

sub _before_execute {
    my ($self, $obj, $param_set) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    ok(!@ids, "No $name found for model.");
    
    return 1;
}

sub _after_execute {
    my ($self, $obj, $param_set) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    ok(@ids, "Added $name to model.");
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Remove::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Genome::Utility::TestBase;
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Remove';
}

sub _valid_param_sets {
    return (
        { # remove multiple 'value_ids' 
            name => 'friends',
            ids => 'Watson,Crick',
        },
        { # remove single 'value'
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
            abandon_builds => 1,
        },
    );
}

sub invalid_param_sets {
    return (
        {# try to remove a not 'is_many' input
            name => 'coolness', 
            ids => 'none',
        },
        { # try to remove input that is not linked to the model anymore
            name => 'inst_data',
            ids => '2sep09.934pmaa1',
        },
    );
}

sub friends {
    return 'Crick';
}

sub inst_data {
    return Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
}

sub _startup {
    my $self = shift;

    my $model = $self->_model;

    $model->add_friend('Crick');
    $model->add_friend('Watson');
    my @friends = $model->friends();
    is_deeply(\@friends, [qw/ Crick Watson /], 'Added friends to remove.');
    my $id = Genome::InstrumentData::Sanger->get('2sep09.934pmaa1')
        or confess "Can't get sange id 2sep09.934pmaa1";
    $model->add_inst_data($id);
    my @inst_data = $model->inst_data;
    is_deeply(\@inst_data, [ $id ], 'Added instr_data to remove');
    
    my $build = $self->_build;
    my $build_input = Genome::Utility::TestBase->create_mock_object(
        class => 'Genome::Model::Build::Input',
        name => 'instrument_data',
        value_id => '2sep09.934pmaa1',
        build_id => $build->id,
    );
    $build->mock('abandon', sub{ return $build->build_event->event_status('Abandoned'); });

    return 1;
}

sub _post_execute {
    my ($self, $obj) = @_;

    my $name = $obj->name;
    my @ids = $self->_model->$name;
    #print Dumper(\@ids);
    ok(!@ids, "Removed $name from model.");
    if ( $obj->abandon_builds ) {
        is($self->_build->status, 'Abandoned', 'Successfully abandoned build');
    }

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::List::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::List';
}

sub valid_param_sets {
    return (
        {
            filter => 'model_id='.$_[0]->_model->id,
        },
    );
}

sub invalid_param_sets {
}

sub _startup {
    my $self = shift;

    $self->_model->coolness('low');
    is($self->_model->coolness, 'low', 'Set coolness (input)');

    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Names::Test;

use strict;
use warnings;

use base 'Genome::Utility::TestCommandBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Names';
}

sub valid_param_sets {
    return (
        {
            type_name => 'tester',
        },
    );
}

sub invalid_param_sets {
    return (
        {
            type_name => 'none',
        },
    );
}

sub startup : Tests(startup) {
    my $self = shift;
    
    no warnings 'once';
    no warnings 'redefine';
    *Genome::Model::Command::get_model_type_names = sub{ 
        return (qw/ tester /); 
    };
    
    return 1;
}

###########################################################################

package Genome::Model::Command::Input::Update::Test;

use strict;
use warnings;

use base 'Genome::Model::Command::Input::TestBase';

use Carp 'confess';
use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::Command::Input::Update';
}

sub _valid_param_sets {
    return (
        { # remove single 'value'
            name => 'coolness',
            value => 'moderate',
        },
        { # remove single 'value'
            name => 'coolness',
            value => 'UNDEF',
        },
    );
}

sub invalid_param_sets {
    return (
        { # try to update an 'is_many' input
            name => 'friends',
            value => 'Watson',
        },
    );
}

###########################################################################

1;

=pod

=head1 Tests

=head1 Disclaimer

 Copyright (C) 2009 Washington University Genome Sequencing Center

 This script is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY
 or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
 License for more details.

=head1 Author(s)

 Eddie Belter <ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

