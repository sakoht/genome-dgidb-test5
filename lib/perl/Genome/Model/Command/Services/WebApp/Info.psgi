#!/usr/bin/env perl

use Web::Simple 'Genome::Model::Command::Services::WebApp::Info';


package Genome::Model::Command::Services::WebApp::Info;

use English;
use Data::Dumper;

sub dispatch_request {
    sub (/**) {
        [
            200,
            [ 'Content-type', 'text/plain' ],
            [ dump_info() ]
        ];
      }, sub () {
        [ 405, [ 'Content-type', 'text/plain' ], ['Method not allowed'] ];
      }
};

Genome::Model::Command::Services::WebApp::Info->run_if_script;


sub dump_info {

    my @inc = map {join(': ',$_,$INC{$_})} sort keys %INC;
    my @env = map {join(': ',$_,$ENV{$_})} sort keys %ENV;

    my $i = join("\n",
        'This page dumps %INC and %ENV for debugging.',
        "\n",
        scalar(localtime()),
        "EXECUTABLE_NAME: " . $^X,
        "PROGRAM NAME: $PROGRAM_NAME",
        "\n***INC***",
        join("\n", @inc),
        "\n\n***ENV***",
        join("\n", @env)
    );

    return $i;
}


