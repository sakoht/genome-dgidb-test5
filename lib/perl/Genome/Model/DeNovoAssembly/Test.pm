package Genome::Model::DeNovoAssembly::Test;

use strict;
use warnings;

require Carp;
use Data::Dumper 'Dumper';
require File::Temp;
require Genome::Model::Test;
require Genome::Utility::TestBase;

BEGIN {
    $ENV{UR_USE_DUMMY_AUTOGENERATED_IDS} = 1;
    $ENV{UR_DBI_NO_COMMIT} = 1;
}

#< Dirs >#
sub _base_directory {
    return '/gsc/var/cache/testsuite/data/Genome-Model/DeNovoAssembly';
}

sub _example_directory_for_model {
    my ($self, $model) = @_;

    Carp::confess "No model to get example directory" unless $model;
    
    my %dirs_versions = (
        soap_solexa => '9',
        velvet_solexa => '0.3',
        abyss_solexa => '1'
    );

    my $assembler_platform = $model->processing_profile->assembler_base_name.'_'.$model->sequencing_platform;
    my $dir = $self->_base_directory.'/'.$assembler_platform.'_build_v'.$dirs_versions{$assembler_platform};

    Carp::confess("Example directory ($dir) for de novo assembly model does not exist.") unless -d $dir;
    
    return $dir;
}

sub _instrument_data_dir {
    return $_[0]->_base_directory.'/inst_data/';
}
#<>#

#< Library Taxon >#
our($taxon, $sample, $library);
sub _create_taxon_sample_and_library {
    my $class = shift;

    $taxon = Genome::Taxon->create(
        name => 'Escherichia coli TEST',
        domain => 'Bacteria',
        current_default_org_prefix => undef,
        estimated_genome_size => 4500000,
        current_genome_refseq_id => undef,
        ncbi_taxon_id => undef,
        ncbi_taxon_species_name => undef,
        species_latin_name => 'Escherichia coli',
        strain_name => 'TEST',
    );
    if ( not $taxon ) {
        Carp::confess('Cannot create taxon');
    }

    $sample = Genome::Sample->create(
        id => -1234,
        name => 'TEST-000',
        taxon_id => $taxon->id,
    );
    if ( not $sample ) {
        Carp::confess('Cannot create sample');
    }

    $library = Genome::Library->create(
        id => -12345,
        name => $sample->name.'-testlibs',
        sample_id => $sample->id,
    );
    if ( not $library ) { 
        Carp::confess('Cannot create library');
    }

    return 1;
}

sub taxon {
    my $class = shift;
    if ( not $taxon ) {
        $class->_create_taxon_sample_and_library;
    }
    return $taxon;
}

sub library {
    my $class = shift;
    if ( not $library ) {
        $class->_create_taxon_sample_and_library;
    }
    return $library;
}
#<>#

#< PP Models and Builds >#
sub processing_profile_for_soap {
    my $class = shift;

    my $pp = Genome::ProcessingProfile::DeNovoAssembly->create(
        name => 'De Novo Assembly Soap Test',
        sequencing_platform => 'solexa',
        assembler_name => 'soap de-novo-assemble',
        assembler_version => '1.04',
        assembler_params => '-kmer_size 31 -resolve_repeats -kmer_frequency_cutoff 1',
        read_processor => 'trimmer bwa-style -trim-qual-level 10 | filter by-length --filter-length 35 | rename illumina-to-pcap',
        post_assemble => 'standard-outputs',
    );

    if ( not $pp ) {
        Carp::confess('Cannot create de novo assembly processing profile for soap');
    }

    return $pp;
}

sub processing_profile_for_velvet {
    my $class = shift;

    my $pp = Genome::ProcessingProfile::DeNovoAssembly->create(
        name => 'De Novo Assembly Velvet Test',
        sequencing_platform => 'solexa',
        coverage => 0.5,#25000,
        assembler_name => 'velvet one-button',
        assembler_version => '0.7.57-64',
        assembler_params => '-hash_sizes 31 33 35 -min_contig_length 100',
        read_processor => 'trimmer by-length -trim-length 10 | rename illumina-to-pcap',
        post_assemble => 'standard-outputs',
    );

    if ( not $pp ) {
        Carp::confess('Cannot create de novo assembly processing profile for velvet');
    }

    return $pp;
}

sub processing_profile_for_abyss {
    my $class = shift;

    my $pp = Genome::ProcessingProfile::DeNovoAssembly->create(
        name => 'De Novo Assembly Abyss Test',
        sequencing_platform => 'solexa',
        assembler_name => 'abyss parallel',
        assembler_version => '1.2.7',
        assembler_params => '-kmer_size 25,31..35 step 2,50 -num_jobs 4',
    );

    if ( not $pp ) {
        Carp::confess('Cannot create de novo assembly processing profile for abyss');
    }

    return $pp;
}

sub _model_for_assembler {
    my ($class, $assembler) = @_;

    Carp::confess('No assembler given to get model') if not $assembler;

    my $taxon = $class->taxon or die; # confesses
    my $pp_method = 'processing_profile_for_'.$assembler;
    my $pp = $class->$pp_method or die; # confesses
    my $tmpdir = File::Temp::tempdir(CLEANUP => 1);
    my $model = Genome::Model::DeNovoAssembly->create(
        processing_profile => $pp,
        subject_name => $taxon->name,
        subject_type => 'species_name',
        center_name => 'WUGC',
        data_directory => $tmpdir,
    );
    if ( not $model ) {
        Carp::confess('Cannot create de novo assembly model for '.$assembler);
    }

    my $instrument_data = $class->solexa_instrument_data;
    if ( not $instrument_data ) {
        Carp::confess('Cannot get solexa instrument data');
    }

    my $add_ok = $model->add_instrument_data($instrument_data);
    if ( not $add_ok ) {
        Carp::confess('Cannot add instrument data to model');
    }

    my @instrument_data = $model->instrument_data;
    if ( not @instrument_data ) {
        Carp::confess('Cannot get instrument data from model');
    }

    return $model;
}

sub model_for_soap {
    return $_[0]->_model_for_assembler('soap');
}

sub model_for_velvet {
    return $_[0]->_model_for_assembler('velvet');
}

sub model_for_abyss {
    return $_[0]->_model_for_assembler('abyss');
}

sub example_build_for_model {
    my ($class, $model) = @_;

    Carp::confess('No de novo assembly model to create example build') if not $model;

    my $dir = $class->_example_directory_for_model($model) or die;
    my $build = Genome::Model::Build->create(
        model=> $model,
        data_directory => $dir,
    );
    if ( not $build ) {
        Carp::confess('Cannot create de novo assembly build');
    }

    $build->status('Succeeded');
    $build->the_master_event->date_completed(UR::Time->now);

    return $build;
}
#<>#

#< Instruement Data >#
our $solexa_instrument_data;
sub solexa_instrument_data {
    my $class = shift;

    return $solexa_instrument_data if $solexa_instrument_data;

    my $library = $class->library or die; # confesses

    my $archive_path = $class->_instrument_data_dir.'/2854709902/sequence_61H93_8_CGATGT.tar.gz';
    if ( not -s $archive_path ) {
        Carp::confess("No archive path in example directory ($archive_path)");
    }

    $solexa_instrument_data = Genome::InstrumentData::Imported->create(
        id => -7777,
        sequencing_platform => 'solexa',
        import_format => 'fastq',
        import_source_name => 'WUGC',
        read_count => 30000,
        read_length => 100,
        subset_name => '8-CGATGT',
        is_paired_end => 1,
        library => $library,
        median_insert_size => 260,# 181, but 260 was used to generate assembly
        sra_sample_id => 'SRS000000',
    );

    if ( not $solexa_instrument_data ) {
        Carp::confess('Cannot create solexa instrument data');
    }

    my $alloc = Genome::Disk::Allocation->__define__(
        owner_id => $solexa_instrument_data->id,
        owner_class_name => $solexa_instrument_data->class,
        disk_group_name => 'info_alignments',
        mount_path => '/gsc/var/cache/testsuite/data/Genome-Model/DeNovoAssembly',
        group_subdirectory => 'inst_data',
        allocation_path => $solexa_instrument_data->id,
        absolute_path => '/gsc/var/cache/testsuite/data/Genome-Model/DeNovoAssembly/inst_data/2854709902',
    );
    if (not $alloc ) {
        Carp::confess('Could not create disk allocation for instrument data');
    }

    if ( not -s $solexa_instrument_data->archive_path ) {
        Carp::confess('Cannot get archive path from instrument data');
    }

    return $solexa_instrument_data;
}
#<>#

1;

