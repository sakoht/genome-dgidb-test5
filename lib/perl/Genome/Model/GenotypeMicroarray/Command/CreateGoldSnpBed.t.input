1	84647761	84647761	T	C	SNP	ref	SNP	ref
5	156323558	156323558	G	G	ref	ref	ref	ref
5	158662525	158662525	T	T	SNP	SNP	SNP	SNP
9	22966592	22966592	A	A	ref	ref	ref	ref
13	76334809	76334809	T	T	SNP	SNP	SNP	SNP
2	224934756	224934756	G	G	ref	ref	ref	ref
8	119481632	119481632	C	C	ref	ref	ref	ref
11	6366366	6366366	G	G	ref	ref	ref	ref
6	169694703	169694703	T	T	ref	ref	ref	ref
7	12953585	12953585	T	T	ref	ref	ref	ref
