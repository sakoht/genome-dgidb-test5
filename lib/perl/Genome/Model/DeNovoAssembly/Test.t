#!/gsc/bin/perl

use strict;
use warnings;

use above "Genome";

use Data::Dumper 'Dumper';
use Test::More;

use_ok('Genome::Model::DeNovoAssembly::Test') or die;

my $rv;

#< PP >#
# fail - no seq platform
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model();
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - get w/o seq platform',
);
# fail - no assembler
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/o assembler',
);
# fail - invalid seq platform/assembler combo
eval{
    $rv = Genome::Model::DeNovoAssembly::Test->get_mock_model(
        assembler_name => 'consed',
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/ invalid seq paltform/assembler combo',
);

# valid velvet solexa
my $velvet_model = Genome::Model::DeNovoAssembly::Test->get_mock_model(
    assembler_name => 'velvet',
    sequencing_platform => 'solexa',
);
ok($velvet_model, 'Got mock DNA model for velvet solexa') or die;
my @inst_data = $velvet_model->instrument_data;
ok(@inst_data, 'Model instrument data');
ok(-s $inst_data[0]->archive_path, 'Solexa archive path exists');

# valid newbler 454
my $newbler_model = Genome::Model::DeNovoAssembly::Test->get_mock_model(
    assembler_name => 'newbler',
    sequencing_platform => 454,
);
ok($newbler_model, 'Got mock DNA model for newbler 454') or die;

done_testing();
exit;

#$HeadURL$
#$Id$
