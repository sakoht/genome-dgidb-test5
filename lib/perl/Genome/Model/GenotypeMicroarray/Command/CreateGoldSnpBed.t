#!/usr/bin/env perl

use above 'Genome';
use Test::More tests => 4;

my $pkg = 'Genome::Model::GenotypeMicroarray::Command::CreateGoldSnpBed';
use_ok($pkg);

my $input = __FILE__ . ".input";
my $expected = __FILE__ . ".expected";
my $tmpdir = File::Temp::tempdir('create-gold-snp-bed-XXXXX', DIR => '/gsc/var/cache/testsuite/running_testsuites/', CLEANUP => 1);
my $output = join('/', $tmpdir, 'output');

my $ref = Genome::Model::Build::ImportedReferenceSequence->get(
    name => "NCBI-human-build36"
    );

my $cmd = $pkg->create(
    input_file => $input,
    output_file => $output,
    reference => $ref,
    );

ok($cmd, 'Created command');
ok($cmd->execute, 'Executed command');

my $diff = Genome::Sys->diff_file_vs_file($output, $expected);
ok(!$diff, 'output matched expected result') or diag("diff results:\n" . $diff);

done_testing();
