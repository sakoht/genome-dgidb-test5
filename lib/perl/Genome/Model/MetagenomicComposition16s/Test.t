#!/gsc/bin/perl

use strict;
use warnings;

use above "Genome";

use Data::Dumper 'Dumper';
use Test::More;

use_ok('Genome::Model::MetagenomicComposition16s::Test') or die;

my $rv;

#< PP >#
# fail - no seq platform
eval{
    $rv = Genome::Model::MetagenomicComposition16s::Test->get_mock_model();
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - get w/o seq platform',
);
# fail - invalid seq platform
eval{
    $rv = Genome::Model::MetagenomicComposition16s::Test->get_mock_model(
        sequencing_platform => 'solexa',
    );
};
diag($@);
ok(
    !$rv && $@,
    'Failed as expected - create w/ invalid sequencing platform',
);

# valid 
my $model = Genome::Model::MetagenomicComposition16s::Test->get_mock_model(
    sequencing_platform => '454',
);
ok($model, 'Got mock DNA model') or die;
my @inst_data = $model->instrument_data;
ok(@inst_data, 'Model instrument data');

done_testing();
exit;

#$HeadURL$
#$Id$
