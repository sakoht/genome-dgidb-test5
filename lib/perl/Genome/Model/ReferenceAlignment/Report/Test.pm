package Genome::Model::ReferenceAlignment::Report::TestBase;

use strict;
use warnings;

use base 'Genome::Utility::TestBase';

use Data::Dumper 'Dumper';
require Genome::Model::Test;
use Test::More;

sub generator {
    return $_[0]->{_object};
}

sub base_params_for_test_class {
    return (
        build_id => $_[0]->_mock_model->last_complete_build->id,
    );
}

sub required_params_for_class {
    return (qw/ build_id /);
}

sub _mock_model {
    my $self = shift;

    unless ( $self->{_mock_model} ) { 
        $self->{_mock_model} = Genome::Model::Test->create_mock_model(
            type_name => 'reference alignment solexa',
        )
            or die;
    }

    return $self->{_mock_model};
}

sub skip_generate { return 0; }

#< Test Methods >#
sub test01_generate_report : Tests {
    my $self = shift;

    if ( $self->skip_generate ) {
        # Skipping generate...too many methods and data to mock.  This was not tested before
        can_ok($self->test_class, '_add_to_report_xml');
    }
    else {
        my $report = $self->generator->generate_report;
        ok($report, 'Generated report');
        #print Dumper($report);
    }
    
    # Save...maybe?  For comparison?
    # $report->name('Saved '.$report->name);
    # ok($self->report->save, 'Saved report');

    return 1;
}

#######################################################################

package Genome::Model::ReferenceAlignment::Report::DbSnpTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::DbSnp';
}

sub params_for_test_class {
    return (
        name => 'Db Snp',
        #override_model_snp_file => '/gsc/var/cache/testsuite/data/Genome-Model-ReferenceAlignment/build-50001/maq_snp_related_metrics/all.snps',
        $_[0]->base_params_for_test_class,
    );
}

#######################################################################

package Genome::Model::ReferenceAlignment::Report::GoldSnpTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::GoldSnp';
}

sub params_for_test_class {
    return (
        name => 'Gold Snp',
        $_[0]->base_params_for_test_class,
    );
}

sub skip_generate { return 1; }

#######################################################################

package Genome::Model::ReferenceAlignment::Report::SolexaStageOneTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub generator {
    return $_[0]->{_object};
}

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::SolexaStageOne';
}

sub params_for_test_class {
    return (
        $_[0]->base_params_for_test_class,
    );
}

sub skip_generate { return 1; }

#######################################################################

package Genome::Model::ReferenceAlignment::Report::SolexaStageTwoTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::SolexaStageTwo';
}

sub params_for_test_class {
    return (
        $_[0]->base_params_for_test_class,
    );
}

sub skip_generate { return 1; }

#######################################################################

package Genome::Model::ReferenceAlignment::Report::MapcheckTest;

use strict;
use warnings;

use base 'Genome::Model::ReferenceAlignment::Report::TestBase';

use Data::Dumper 'Dumper';
use Test::More;

sub test_class {
    return 'Genome::Model::ReferenceAlignment::Report::Mapcheck';
}

sub params_for_test_class {
    return (
        $_[0]->base_params_for_test_class,
    );
}

sub skip_generate { return 1; }

#######################################################################

1;

=pod

=head1 Name

ModuleTemplate

=head1 Synopsis

=head1 Usage

=head1 Methods

=head2 

=over

=item I<Synopsis>

=item I<Arguments>

=item I<Returns>

=back

=head1 See Also

=head1 Disclaimer

Copyright (C) 2005 - 2008 Washington University Genome Sequencing Center

This module is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY or the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.

=head1 Author(s)

B<Eddie Belter> I<ebelter@watson.wustl.edu>

=cut

#$HeadURL$
#$Id$

