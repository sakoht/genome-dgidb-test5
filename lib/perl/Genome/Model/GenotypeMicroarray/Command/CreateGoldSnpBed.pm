package Genome::Model::GenotypeMicroarray::Command::CreateGoldSnpBed;

use strict;
use warnings;

use Genome;
use Genome::Info::IUB;

class Genome::Model::GenotypeMicroarray::Command::CreateGoldSnpBed {
    is => 'Genome::Command::Base',
    doc => 'Convert the genotype microarray file to the .bed format used by the GoldSnp view',
    has_input => [
        input_file => {
            is => 'File',
            doc => 'The input file Genotype Microarray file to convert.',
            shell_args_position => 1,
        },
        output_file => {
            is => 'File',
            doc => 'The output BED file to write.',
            shell_args_position => 2,
        },
        reference => {
            is => 'Genome::Model::Build::ImportedReferenceSequence',
            id_by => 'reference_id',
            doc => 'The reference sequence build id to use',
            shell_args_position => 3,
        },
        reference_id => {
            is => 'Integer',
        },
    ],
};

sub execute {
    my $self = shift;
    $self->{bases_files} = {};

    # Unfortunately, some of these genotype files seem to be unsorted.
    # It would be better sort first, so the reference sequence can be
    # accessed sequentially. However, joinx sort can't deal with -- in the
    # stop field of the input, which seems to happen sometimes, so we let 
    # _convert_file filter those out then sort after.

    my $tmpfile = Genome::Sys->create_temp_file_path;
    my $infile = $self->input_file;
    my $ifh = new IO::File("<$infile") || die "Failed to open input file $infile";
    my $ofh = new IO::File(">$tmpfile") || die "Failed to open output file $tmpfile";
    my $rv = $self->_convert_file($self->reference, $ifh, $ofh);
    $ifh->close();
    $ofh->close();

    my $sort_cmd = Genome::Model::Tools::Joinx::Sort->create(
        input_files => [$tmpfile],
        output_file => $self->output_file,
    );
    if (!$sort_cmd->execute) {
        $self->error_message("Failed to sort input genotype file " . $self->input_file);
        return;
    }

    return $rv;
}

sub _convert_file {
    my ($self, $reference, $ifh, $ofh) = @_;

    while (my $line = <$ifh>) {
        chomp $line;
        my ($chr, $start, $end, $allele1, $allele2, $a1t1, $a2t1, $a1t2, $a2t2) = split("\t", $line);

        if ($end =~ /^-/) {
            $self->warning_message("Invalid end position '$end', skipping");
            next;
        }

        my $ref = $reference->sequence($chr, $end, $end); 
        if (!$ref) {
            $self->warning_message("No reference for position $chr $end $end, skipping $line\n");
            next;
        }
        $ref = uc($ref);

        if ($allele1 eq $allele2) {
            if ($a1t1 ne $a2t1 or $a1t2 ne $a2t2) {
                print STDERR "Inconsistent types within a platform $line\n";
                next;
            }

            if ($a1t1 eq 'ref' and $allele1 ne $ref) {
                print STDERR "Reference mismatch at $chr $end: expected $ref, got $allele1, line: $line\n";
                next;
            }

            if ($a1t1 ne 'ref' and $allele1 eq $ref) {
                print STDERR "Gold SNP is listed as reference in B36 sequence: $line\n";
                next;
            }
        } else {
            if ($a1t1 eq $a2t1) {
                print STDERR "Het snp where both alleles are non-ref: $line\n";
                next;
            }

            if ($a1t1 ne 'SNP' and $a2t1 ne 'SNP') {
                print STDERR "Supposedly het snp not labeled as such: $line\n";
                next;
            }
        }

        my $iub = Genome::Info::IUB::iub_for_alleles($allele1, $allele2);

        my $new_call = "$ref/$iub";
        --$start;
        my $score = 0;
        my $depth = 0;
        $ofh->print(join("\t", $chr, $start, $end, $new_call, $score, $depth, $a1t1, $a2t1, $a1t2, $a2t2) . "\n");

    }
    $ifh->close();
    $ofh->close();
    return 1;
}

1;
