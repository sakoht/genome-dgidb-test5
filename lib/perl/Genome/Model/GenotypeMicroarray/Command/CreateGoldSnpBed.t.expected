1	84647760	84647761	C/Y	0	0	SNP	ref	SNP	ref
2	224934755	224934756	G/G	0	0	ref	ref	ref	ref
5	156323557	156323558	G/G	0	0	ref	ref	ref	ref
5	158662524	158662525	C/T	0	0	SNP	SNP	SNP	SNP
6	169694702	169694703	T/T	0	0	ref	ref	ref	ref
7	12953584	12953585	T/T	0	0	ref	ref	ref	ref
8	119481631	119481632	C/C	0	0	ref	ref	ref	ref
9	22966591	22966592	A/A	0	0	ref	ref	ref	ref
11	6366365	6366366	G/G	0	0	ref	ref	ref	ref
13	76334808	76334809	G/T	0	0	SNP	SNP	SNP	SNP
